lib/rules/logical-assignment-operators.js

```diff
@@ -0,0 +1,430 @@
+                                                        (astUtils.getPrecedence({ type: "AssignmentExpression" }) < astUtils.getPrecedence(logical.parent));
+
+                            if (!astUtils.isParenthesised(sourceCode, logical) && requiresParenthesis) {
+                                yield ruleFixer.insertTextBefore(logical, "(");
+                                yield ruleFixer.insertTextAfter(logical, ")");
+                            }
+
+                            // Right side has to be parenthesized, otherwise would be parsed as (foo || foo) = bar which is illegal
+                            const leftToStartRight = [logical.range[0], logical.right.range[0]];
+
+                            yield ruleFixer.removeRange(leftToStartRight); // -> foo = bar)
+
+                            const endParenthesis = sourceCode.getTokenAfter(logical.right);
+
+                            yield ruleFixer.remove(endParenthesis); // -> foo = bar
+
+                            const operatorToken = getOperatorToken(logical.right);
+
+                            yield ruleFixer.insertTextBefore(operatorToken, logical.operator); // -> foo ||= bar
+                        }
+                    });
+                }
+            },
+
+            // if (foo) foo = bar
+            "IfStatement[alternate=null]"(ifNode) {
+                if (!checkIf) {
+                    return;
+                }
+
+                const hasBody = ifNode.consequent.type === "BlockStatement";
+
+                if (hasBody && ifNode.consequent.body.length !== 1) {
+                    return;
+                }
+
+                const body = hasBody ? ifNode.consequent.body[0] : ifNode.consequent;
+                const scope = context.getScope();
+                const existence = getExistence(ifNode.test, scope);
+
+                if (
+                    body.type === "ExpressionStatement" &&
+                    body.expression.type === "AssignmentExpression" &&
+                    body.expression.operator === "=" &&
+                    existence !== null &&
+                    astUtils.isSameReference(existence.reference, body.expression.left)
+                ) {
+                    const descriptor = {
+                        messageId: "if",
+                        node: ifNode
+                    };
+                    const suggestion = {
+                        messageId: "convertIf",
+                        *fix(ruleFixer) {
+                            if (sourceCode.getCommentsInside(ifNode).length > 0) {
+                                return;
+                            }
+
+                            const operatorToken = getOperatorToken(body.expression);
+                            const prevToken = sourceCode.getTokenBefore(ifNode);
+
+                            if (astUtils.isParenthesised(sourceCode, body.expression.left) && (prevToken !== null && !astUtils.isSemicolonToken(prevToken))) {
```

Comment:
There are more cases where `body` can start with `(`, for example:

```js
/* eslint logical-assignment-operators: ["error", "always", { enforceForIfStatements: true }] */

foo()
if (a.b == null) {
    (a).b = 5;
}

```

After the fix:

```js
/* eslint logical-assignment-operators: ["error", "always", { enforceForIfStatements: true }] */

foo()
(a).b ??= 5;
```

This would be parsed as `foo()(a).b ??= 5;`.

I think the solution I suggested in https://github.com/eslint/eslint/pull/16102#discussion_r917451987 is a safe way to handle these rare cases (+ we can autofix/suggest if there is no previous token or if the previous token is `;`).

Reply:
I included `{` for the previous token check to also allow the fix/suggestion when the `if` is the first statement in the block